src/keeper.ts

```diff
@@ -167,7 +172,26 @@ class Keeper {
         );
       })
     );
-    return nestedEvents.flat(1);
+    // warn about requesting too many events
+    nestedEvents.map((singleFilterEvents, index) => {
+      if (singleFilterEvents.length > 1000) {
```

Comment:
Do we want the warning at 1000 if we only need to start batching after 10000?

Reply:
I am not sure what will happen at 10000, I think it's possible that it will either start crashing, or start reporting wrong data, so the warning is just an attempt to do something before that happens (but that I didn't have time to deal with this time).

I think batching by position id range should be not too complicated to intoduce, I've created a follow up issue for this.